CMakeLists.txt

```diff
@@ -1,64 +1,56 @@
 cmake_minimum_required(VERSION 3.10) # CMP0083 NEW
 
 include(CMake/out_of_tree.cmake)
+include(CMake/genex.cmake)
 
-if(NOT CMAKE_BUILD_TYPE)
-  set(CMAKE_BUILD_TYPE "Debug")
-endif()
-
-option(ASAN "Enable address sanitizer" ON)
-option(UBSAN "Enable undefined behaviour sanitizer" ON)
-option(DEBUG "Enable debug mode in engine" ON)
+DEBUG_OPTION(ASAN "Enable address sanitizer")
+DEBUG_OPTION(UBSAN "Enable undefined behaviour sanitizer")
+DEBUG_OPTION(DEBUG "Enable debug mode in engine")
+option(DISABLE_LTO "Disable link-time optimization (by default enabled in release mode)" OFF)
 option(PIE "Generate position-independent code" OFF)
-option(LTO "Enable link-time optimization (if supported by the toolchain)" OFF)
 option(SPAWN "Build the shareware version" OFF)
 option(DIST "Dynamically link only glibc and SDL2" OFF)
-option(FASTER "Enable FASTER in engine" ON)
 option(BINARY_RELEASE "Enable options for binary release" OFF)
 option(NIGHTLY_BUILD "Enable options for nightly build" OFF)
 option(USE_SDL1 "Use SDL1.2 instead of SDL2" OFF)
 option(NONET "Disable network" OFF)
 option(RUN_TESTS "Build and run tests" OFF)
 
-if(BINARY_RELEASE)
+if(BINARY_RELEASE OR CMAKE_BUILD_TYPE STREQUAL "Release")

```

Comment:
Good idea to make this work with the CMAKE_BUILD_TYPE

Reply:
Yeah, many package managers set it by default (e.g. buildroot) so this way we'll get fast packages even if the person making the package didn't read build instructions (totally a thing).